sympy/parsing/sympy_parser.py

```diff
@@ -983,6 +983,30 @@ def flatten(self, args, func):
                 result.append(arg)
         return result
 
+    def visit_Call(self, node):
+        from sympy import sympify
+        from sympy.core.expr import Expr, AtomicExpr
+        node_class = sympify(node.func.id)
+        if isinstance(node_class, type) and issubclass(node_class, Basic) and not issubclass(node_class, AtomicExpr):
+            args = []
+            for arg in node.args:
+                if isinstance(arg, ast.BinOp):
```

Comment:
The docs say one should use `generic_visit` to handle subnodes. Also, does the visit_BinOp do this correctly?

Reply:
using
```
if isinstance(arg, (ast.BinOp, ast.Call)):
                    args.append(self.generic_visit(arg))
```
doesnot stop evaluation of subnodes.